scripts/graph-deploy.sh

```
@@ -0,0 +1,53 @@
+yq -i "(.dataSources[] | select(.name == \"SystemFactory\").source.address) = \"$SYSTEM_FACTORY_ADDRESS\"" ./subgraph/subgraph.yaml
+
+# Print addresses for debugging
+echo "---"
+echo "Addresses being used:"
+echo "  SMARTSystemFactory: $SYSTEM_FACTORY_ADDRESS"
+echo "---"
+echo
+
+if [ "$DEPLOY_ENV" == "local" ]; then
+npx graph create --node http://localhost:8020 smart
+npx graph deploy --version-label v1.0.$(date +%s) --node http://localhost:8020 --ipfs https://ipfs.console.settlemint.com smart ./subgraph/subgraph.yaml
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)